### ISSUE TITLE
No JS works with React 18

### ISSUE THREAD

**Issue by MRezaSafari** [thanhlmm/next-lazy-hydrate#3]
## Current Behavior

When using lazy hydrate not a single line of js getting executed in component

## Expected Behavior

JS also should work

## Steps to Reproduce the Problem

  1. create a simple component with something like onHover or onMouseEnter or event an alert in useEffect
  2. use withLazyHydrate
  3. no js will work 

## Environment

- Version: 1.1.0
- Platform: Mac
- Node.js Version: 16.16.0

**Comment by jimmi-rbn** [comment]
BUMP

**Comment by thanhlmm** [comment]
Hey, sorry for the late update, I missed this notification.
Basically, if you're using React 18, you don't need this. React 18 support Lazy Hydratate out of the box, the missing thing is lazyload js which I think I gonna update it soon

**Comment by julpat** [comment]
According to this issue, I thought it should not work at all with React 18+ but it is (hydration + javascript inside the component).
It's working but together with this React "legacy" warning

`Warning: ReactDOM.hydrate is no longer supported in React 18. Use hydrateRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot`

**Comment by thanhlmm** [comment]
You're right, as my last comment, if you use react 18 you don't need this

**Comment by julpat** [comment]
But if I want to achieve "lazy hydrate when in view" I need your solution. I didn't find any other with server-side support. Or am I missing something?

**Comment by thanhlmm** [comment]
hmm, maybe we can do some trick with react suspense, but not for right. I will try that solution when I have time

**Comment by MRezaSafari** [comment]
any update about this?

**Comment by tunaayberk** [comment]
Tested on next 13.3.0 with react 18.2.0

Only things I updated are in Line 103 - 104 of `lazy-hydrate.tsx` file. 

Updating those lines with following code will fix problem:
```
const action = el.hasChildNodes() ? "hydrateRoot" : "render";
ReactDOM[action](el, React.createElement(result, props));
```
This will replace `hydrate` function with `hydrateRoot` and change order of arguments which is only difference of those functions.

This should resolve the issue and work with React 18.

**Comment by MurkyMeow** [comment]
I've made an example that works in Next 13/14 https://github.com/Fractal-Web/next-ssr-hydration
